cmd/subctl/join.go

```diff
@@ -0,0 +1,238 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package subctl
+
+import (
+	"errors"
+	"fmt"
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/spf13/cobra"
+	submariner "github.com/submariner-io/submariner-operator/api/submariner/v1alpha1"
+	"github.com/submariner-io/submariner-operator/internal/cli"
+	"github.com/submariner-io/submariner-operator/internal/exit"
+	"github.com/submariner-io/submariner-operator/internal/restconfig"
+	"github.com/submariner-io/submariner-operator/pkg/broker"
+	"github.com/submariner-io/submariner-operator/pkg/join"
+	"github.com/submariner-io/submariner-operator/pkg/subctl/datafile"
+	"strings"
+)
+
+var joinFlags join.Options
+
+var joinCmd = &cobra.Command{
+	Use:     "join",
+	Short:   "Connect a cluster to an existing broker",
+	Args:    cobra.MaximumNArgs(1),
+	PreRunE: restConfigProducer.CheckVersionMismatch,
+	Run: func(cmd *cobra.Command, args []string) {
+		err := checkArgumentPassed(args)
+		exit.OnError(err)
+
+		subctlData, err := datafile.NewFromFile(args[0])
+		exit.WithMessage("Error loading the broker information from the given file", err)
+		fmt.Printf("* %s says broker is at: %s\n", args[0], subctlData.BrokerURL)
```

Comment:
This should use the new `broker.ReadInfoFromFile`. Also let's use the CLI status here:

 ```suggestion
                 status := cli.NewReporter()
                 
                 brokerInfo, err := broker.ReadInfoFromFile(args[0])
                 exit.OnError(status.Error(err, "Error loading the broker information from file %q", args[0]))

                 status.Success("Loaded the broker information from file %q", args[0])
                 status.Success("The broker is located at %q", brokerInfo.BrokerURL)
```

Reply:
Commit https://github.com/submariner-io/submariner-operator/pull/1726/commits/6fc60d5e759dbebb4f84a4806616706bfaa5de6a for new `broker.ReadInfoFromFile` and https://github.com/submariner-io/submariner-operator/pull/1726/commits/e05535f023bd0dd66743675e987261a0b36c17a3 for `reporter.Interface`